projects/zxing-scanner/src/lib/zxing-scanner.component.ts

```diff
@@ -453,7 +453,8 @@ export class ZXingScannerComponent implements AfterViewInit, OnDestroy {
       // Asks for permission before enumerating devices so it can get all the device's info
       hasPermission = await this.askForPermission();
     } catch (e) {
-      console.error(e);
+      console.error('Exception occurred while asking for permission:', e);
+      return;
```

Comment:
You sure an explicit void return is needed here?

```suggestion
```

Reply:
It's not needed since the hasPermission will not evaluate to true afterwards, but I thought it's cleaner to exit the function at this point.

Reply:
I don't know why, but I tought JS would exit the function to the outer scope after an `Error` being caught in a `catch` block. Since that's not the real behavior, you're totally right, the `return` should be there.